### PR TITLE
Fix/misc homepage ui issues and minor general bug fixes

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR fixes some misc homepage ui issues, to resolve https://github.com/isomerpages/isomercms-frontend/issues/232.

**Bug fixes**
---
- This PR also fixes a bug involving the saving of files in PR https://github.com/isomerpages/isomercms-frontend/pull/219 - I accidentally changed one of the arguments when generating collection file names, which broke the flow for collection pages. 
- It also fixes an existing bug where renaming 3rd nav pages was not being handled.
- Also, it fixes a bug where clicking on the text to open the ComponentSettingsModal in OverviewCard would not work.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -333,6 +331,10 @@ const generateNewCollectionFileName = async ({
 
       newFileName = generateCollectionPageFileName(title, groupIdentifier);
     }
+  } else if (originalThirdNavTitle === thirdNavTitle && collectionPageData) {
+    // Case: renaming/changing details of existing file in collection
+    const groupIdentifier = await generateGroupIdentifier(collectionPageData, thirdNavTitle ? true : false, baseApiUrl, false, thirdNavTitle)
+    newFileName = generateCollectionPageFileName(title, groupIdentifier);
```
Comment (kwajiehao) [inline src/utils.js:337]: This fails when trying to rename a collection third nav page to the same third nav section (for example, if I want to rename a page with title `abc` to `def` but to keep it within the same third nav section, `5a-abc.md` should become `5a-def.md`). Currently, it returns `undefined`, or it generates a non-third-nav group identifier by adding it as the last page to the collection (for example, `8-abc.md`)

Comment (kwajiehao) [reply src/utils.js:337]: I added the following two cases in the excel sheet https://docs.google.com/spreadsheets/d/1-U69XpMmSHwtOXSYkWGc5gHpwrvA4i23cSQgOA4j5_s/edit#gid=2010487752:

- Rename an existing third nav page while keeping it in the same third nav section
- Rename an existing (normal) collection page while keeping it as a normal collection page

**Review by kwajiehao — CHANGES_REQUESTED** [review]
Left comments

One more issue: currently, the templates for `infopic-left` and `infopic-right` are wrong. The subtitle should come above the title for the templates to remain faithful to the original Isomer templates. So what we need to do is:
- correct the templates to switch `title` and `subtitle` for `infopic-left` and `infopic-right`
- correct `EditHomepage` so that we switch `title` and `subtitle` fields for `infopic-left` and `infopic-right`

File: src/utils.js (hunk 1 above)
Comment (alexanderleegs) [reply src/utils.js]: Good catch, we should just reuse the existing group identifier in these cases. Resolved in https://github.com/isomerpages/isomercms-frontend/pull/246/commits/4c310d2136e2a1242c4df3e398e65a2c8a98cbba

**Comment by alexanderleegs** [comment]
> One more issue: currently, the templates for `infopic-left` and `infopic-right` are wrong. The subtitle should come above the title for the templates to remain faithful to the original Isomer templates. So what we need to do is:
> 
> * correct the templates to switch `title` and `subtitle` for `infopic-left` and `infopic-right`
> * correct `EditHomepage` so that we switch `title` and `subtitle` fields for `infopic-left` and `infopic-right`

Resolved in https://github.com/isomerpages/isomercms-frontend/pull/246/commits/8fdbf1017f7aa235297fce5563f3d31681cb7f0b

**Review by kwajiehao — APPROVED** [review]
lgtm

**Pull request merged**